ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            pszDesc = MMReturnValueFromSectionINIFile(pszRelFile, section,
+                                                      "descriptor_cat");
+            if (pszDesc)
+            {
+                pszString = CPLRecode_function(pszDesc, CPL_ENC_ISO8859_1,
+                                               CPL_ENC_UTF8);
+                free_function(pszDesc);
+
+                MM_strnzcpy(
+                    pMMBDXP->pField[nIField].FieldDescription[MM_SPA_LANGUAGE],
+                    pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+
+                if (*pMMBDXP->pField[nIField]
+                         .FieldDescription[MM_DEF_LANGUAGE] != '\0')
+                {
+                    MM_strnzcpy(pMMBDXP->pField[nIField]
+                                    .FieldDescription[MM_DEF_LANGUAGE],
+                                pszString, MM_MAX_LON_DESCRIPCIO_CAMP_DBF);
+                }
+
+                free_function(pszString);
+            }
+            else
+                *pMMBDXP->pField[nIField].FieldDescription[MM_SPA_LANGUAGE] =
+                    '\0';
+        }
+    }
+
+    if (!pMMBDXP->nFields)
+    {
+        if (pMMBDXP->BytesPerRecord)
+            incoherent_record_size = TRUE;
+    }
+    else if (pMMBDXP->pField[pMMBDXP->nFields - 1].BytesPerField +
+                 pMMBDXP->pField[pMMBDXP->nFields - 1].AcumulatedBytes >
+             pMMBDXP->BytesPerRecord)
+        incoherent_record_size = TRUE;
+    if (incoherent_record_size)
+    {
+        if (n_queixes_estructura_incorrecta == 0)
+        {
+            incoherent_record_size = FALSE;
+            fseek_function(pf, offset_reintent, SEEK_SET);
+            n_queixes_estructura_incorrecta++;
+            goto reintenta_lectura_per_si_error_CreaCampBD_XP;
+        }
+    }
+
+    offset_possible = 32 + 32 * (pMMBDXP->nFields) + 1;
+
+    if (!incoherent_record_size &&
+        offset_possible != pMMBDXP->FirstRecordOffset)
+    {  // Extended names
+        MM_FIRST_RECORD_OFFSET_TYPE offset_nom_camp;
+        int mida_nom;
+
+        for (nIField = 0; nIField < pMMBDXP->nFields; nIField++)
+        {
+            offset_nom_camp =
+                MM_GiveOffsetExtendedFieldName(pMMBDXP->pField + nIField);
+            mida_nom = MM_DonaBytesNomEstesCamp(pMMBDXP->pField + nIField);
+            if (mida_nom > 0 && mida_nom < MM_MAX_LON_FIELD_NAME_DBF &&
+                offset_nom_camp >= offset_possible &&
+                offset_nom_camp < pMMBDXP->FirstRecordOffset)
+            {
+                MM_strnzcpy(pMMBDXP->pField[nIField].ClassicalDBFFieldName,
+                            pMMBDXP->pField[nIField].FieldName,
+                            MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF);
+                fseek_function(pf, offset_nom_camp, SEEK_SET);
+                if (1 != fread_function(pMMBDXP->pField[nIField].FieldName,
+                                        mida_nom, 1, pf))
+                {
+                    free(pMMBDXP->pField);
```

Comment:
same remark as above regarding cleanup of pField

Reply:
done